DSCResources/MSFT_xADRequestODJ/MSFT_xADRequestODJ.psm1

```diff
@@ -0,0 +1,321 @@
+$moduleRoot = Split-Path `
+    -Path $MyInvocation.MyCommand.Path `
+    -Parent
+
+#region LocalizedData
+$Culture = 'en-us'
+if (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath $PSUICulture))
+{
+    $Culture = $PSUICulture
+}
+Import-LocalizedData `
+    -BindingVariable LocalizedData `
+    -Filename MSFT_xADRequestODJ.psd1 `
+    -BaseDirectory $moduleRoot `
+    -UICulture $Culture
+#endregion
+
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $DomainName,
+
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $ComputerName,
+
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $RequestFile
+    )
+
+    Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+        $($LocalizedData.GettingRequestODJMessage)
+        ) -join '')
+
+    Assert-Module -ModuleName 'ActiveDirectory'
+
+    # It is not possible to determine the state of the ODJ except for the OU
+    # So most parameters are not returned.
+    $returnValue = @{
+        DomainName = $DomainName
+        ComputerName = $ComputerName
+        RequestFile = $RequestFile
+    }
+
+    $Domain = Get-ADDomain -Identity $DomainName
+    $Parameters = @{
+        searchbase = $Domain.DistinguishedName
+        filter = { name -eq $ComputerName }
+    }
+
+    $Computer = Get-ADComputer @Parameters
+    if ($Computer)
+    {
+        $returnValue += @{
+            OU = ($Computer.DistinguishedName -replace "CN=$ComputerName,",'')
+        }
+    } # if
+
+    #Output the target resource
+    $returnValue
+} # Get-TargetResource
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $DomainName,
+
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $ComputerName,
+
+        [System.String]
+        $OU,
+
+        [System.String]
+        $DCName,
```

Comment:
For consistency with the other resources, should we name this to `DomainController`? What about credentials (assuming WMF 4)?